tests/testsuite/publish.rs

```diff
@@ -961,7 +961,6 @@ fn publish_failed_with_index_and_only_allowed_registry() {
         .arg(registry.index_url().as_str())
         .with_status(101)
         .with_stderr_data(str![[r#"
-[NOTE] found `alternative` as only allowed registry. Publishing to it automatically.
```

Comment:
Why was the note removed?

Reply:
`infer_registry` doesn't print this warning if `--index` or `--registry` was provided. The warning is misleading: we didn't automatically choose where to publish.

Reply:
Thanks for that fix!